xmtp_mls/src/groups/device_sync/backup.rs

```diff
@@ -0,0 +1,180 @@
+use super::DeviceSyncError;
+use crate::storage::xmtp_openmls_provider::XmtpOpenMlsProvider;
+use backup_exporter::BackupExporter;
+use std::{path::Path, sync::Arc};
+use thiserror::Error;
+use xmtp_common::time::now_ns;
+use xmtp_proto::xmtp::device_sync::{BackupElementSelection, BackupMetadata};
+
+pub use backup_importer::BackupImporter;
+
+// Increment on breaking changes
+const BACKUP_VERSION: u32 = 0;
+
+mod backup_exporter;
+mod backup_importer;
+mod export_stream;
+
+#[derive(Debug, Error)]
+pub enum BackupError {
+    #[error("Missing metadata")]
+    MissingMetadata,
+}
+
+pub struct BackupOptions {
+    pub start_ns: Option<i64>,
+    pub end_ns: Option<i64>,
+    pub elements: Vec<BackupElementSelection>,
+}
+
+impl From<BackupOptions> for BackupMetadata {
+    fn from(value: BackupOptions) -> Self {
+        Self {
+            backup_version: BACKUP_VERSION,
+            end_ns: value.end_ns,
+            start_ns: value.start_ns,
+            elements: value.elements.iter().map(|&e| e as i32).collect(),
+            exported_at_ns: now_ns(),
+        }
+    }
+}
+
+impl BackupOptions {
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
what's the plan for web support/how would i use this from the web?

Reply:
Just like how we have a handler that consumes the `AsyncRead` to a file for native [here](https://github.com/xmtp/libxmtp/pull/1479/files#diff-d276d35d0e0e31d58f70cacfa50d236b16e8e5571aaed8b08c2f7d624210c924R11), we'd want to do something similar for wasm files.

Reply:
is it possible to get a test that just reads bytes from the exported backup from wasm?